expcore/auto_play.lua

```diff
@@ -0,0 +1,519 @@
+            {"", "Forest Safari race"},
+        }
+    },
+    race_fuel = {
+        title = "Fuel",
+        items = {
+            {"", "[img=item/wood] ", {"item-name.wood"}},
+            {"", "[img=item/solid-fuel] ", {"item-name.solid-fuel"}},
+            {"", "[img=item/rocket-fuel] ", {"item-name.rocket-fuel"}},
+            {"", "[img=item/nuclear-fuel] ", {"item-name.nuclear-fuel"}}
+        },
+        values = { "wood", "solid-fuel", "rocket-fuel", "nuclear-fuel" }
+    },
+    race_laps = {
+        title = "Laps",
+        items = {
+            {"", "1 Lap"},
+            {"", "2 Laps"},
+            {"", "3 Laps"},
+        }
+    }
+}
+
+--- Globals
+local primitives = { enabled = false, game_running = false }
+local player_data = {}
+
+--- Register the global variables
+Global.register({
+    primitives = primitives,
+    player_data = player_data,
+},function(tbl)
+    primitives = tbl.primitives
+    player_data = tbl.player_data
+end)
+
+--- Add public API to /interface command
+Interface.add_interface_module('Auto_Play', Auto_Play)
+
+
+-- Updates the display of the welcome screen and the voting screen
+-- Does not update vote counts
+local function update_selector_window(player)
+    local role_window = player.gui.center.role_selector_window
+    local vote_window = player.gui.screen.race_vote_window
+    if primitives.game_running or not primitives.enabled then
+        role_window.visible = false
+        vote_window.visible = false
+        return
+    end
+
+    if player_data[player.index].show_welcome_screen then
+        role_window.visible = true
+        vote_window.visible = false
+    else
+        role_window.visible = false
+        vote_window.visible = true
+
+        local participant = Roles.player_has_role(player, 'Participant')
+        local data = player_data[player.index]
+        vote_window.caption = participant and "Joining Race" or "Spectating Race"
+        data.toggle_participant.caption = participant and "Spectate Game Instead" or "Join Game Instead"
+    end
+end
+
+--- Welcome screen spectate button
+local SpectateButton =
+Gui.element{
+    type = 'button',
+    caption = "Spectate Game",
+    style = 'button',
+}
+:on_click(function(player, _)
+    -- Remove welcome screen and remove player from participants
+    player_data[player.index].show_welcome_screen = false
+    Roles.unassign_player(player, {'Participant'}, nil, true, true)
+    update_selector_window(player)
+end)
+
+--- Welcome screen join button
+local JoinButton =
+Gui.element{
+    type = 'button',
+    caption = "Join Game",
+    style = 'green_button',
+}
+:on_click(function(player, _, event)
+    -- Remove welcome screen and add player to participants
+    player_data[event.player_index].show_welcome_screen = false
+    Roles.assign_player(player, {'Participant'}, nil, true, true)
+    update_selector_window(player)
+end)
+
+--- Toggle participation button for race vote screen
+local ToggleParticipant =
```

Comment:
Could you make this button horizontally stretchable.
Can be done with `:style{ horizontally_stretchable = true }`